sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1048,6 +1048,9 @@ static int get_dailink_info(struct device *dev,
 				return -EINVAL;
 			}
 
+			if (adr_link->adr_d[i].num_endpoints > 1)
+				dev_warn(dev, "Multiple endpoints not currently supported\n");
```

Comment:
i think we changed this recently, didn't we? @bardliao added support for RT712 with 2 endpoints.

Reply:
Hmm... it appears he did this patch might be bogus, I will double check.

Reply:
I can see that rt712 was added with 2 endpoints, but as far as I can see no part of the machine driver accesses anything other than the first endpoint. My assumption is this is still a work in progress and whilst the endpoints are defined at the moment no support is added for the second one. I am happy to drop this patch if support is incoming shortly?

Reply:
Indeed, we only check the first endpoint in machine driver. Currently, we only care about if the endpoint is aggregated or not. For those codecs with multiple endpoints, we expect they are all non-aggregated. Maybe show the warning message only when it is with multiple endpoints, and any of the endpoint is aggregated?
Does it make sense? @charleskeepax @plbossart

Reply:
the concept of endpoint was a placeholder for future ACPI-based descriptions on the hardware. Indeed if a codec exposes multiple endpoints, the idea is that they can be used independently for different usages. The problem is that ACPI work has been put on hold for what looks like forever and I am not sure there's any merit in trying to improve these endpoints. Better to leave them as is and maybe we'll remove all of this at some point.